Python/Product/Analysis/LanguageServer/Server.cs

```diff
@@ -82,8 +86,15 @@ public sealed class Server : ServerBase, IDisposable {
 
         #region Client message handling
 
-        public async override Task<InitializeResult> Initialize(InitializeParams @params) {
-            _analyzer = await CreateAnalyzer(@params.initializationOptions.interpreter);
+        public override async Task<InitializeResult> Initialize(InitializeParams @params) {
+            _testEnvironment = @params.initializationOptions.interpreter.properties.ContainsKey("TestEnvironment");
+            if (_testEnvironment) {
+                // Test environment needs predictable initialization.
+                // Tests can only proceed when analysis is fully done.
+                _analyzer = CreateAnalyzer(@params.initializationOptions.interpreter).Result;
```

Comment:
Can't the tests `await Initialize` to get the same result?

Reply:
Those tests are in typescript...

Reply:
Then can you move the _testEnvironment variable to the VSC wrapper, rather than embedding it in the server? I've tried really hard to keep this kind of state out of the analyzer this time around :)